plugin/evm/vm_test.go

```diff
@@ -4091,35 +4093,46 @@ func TestParentBeaconRootBlock(t *testing.T) {
 }
 
 func TestNoBlobsAllowed(t *testing.T) {
+	ctx := context.Background()
 	require := require.New(t)
-	importAmount := uint64(1000000000)
-	issuer, vm, _, _, _ := GenesisVMWithUTXOs(t, true, genesisJSONCancun, "", "", map[ids.ShortID]uint64{
-		testShortIDAddrs[0]: importAmount,
-	})
-	defer func() { require.NoError(vm.Shutdown(context.Background())) }()
 
-	// Build a valid block in Cancun
-	importTx, err := vm.newImportTx(vm.ctx.XChainID, testEthAddrs[0], initialBaseFee, []*secp256k1.PrivateKey{testKeys[0]})
-	require.NoError(err)
-	require.NoError(vm.mempool.AddLocalTx(importTx))
-	<-issuer
-	blk, err := vm.BuildBlock(context.Background())
+	gspec := new(core.Genesis)
+	err := json.Unmarshal([]byte(genesisJSONCancun), gspec)
 	require.NoError(err)
 
-	// Modify the block to have a non-zero blob gas
-	ethBlock := blk.(*chain.BlockWrapper).Block.(*Block).ethBlock
-	header := types.CopyHeader(ethBlock.Header())
-	blobGasUsed := uint64(params.BlobTxBlobGasPerBlob)
-	header.BlobGasUsed = &blobGasUsed
-	modifiedEthBlock := types.NewBlockWithExtData(
-		header, nil, nil, nil, new(trie.Trie), ethBlock.ExtData(), false)
-	modifiedBlock, err := vm.newBlock(modifiedEthBlock)
+	// Make one block with a single blob tx
+	signer := types.NewCancunSigner(gspec.Config.ChainID)
+	blockGen := func(_ int, b *core.BlockGen) {
+		b.SetCoinbase(constants.BlackholeAddr)
+		fee := big.NewInt(500)
+		fee.Add(fee, b.BaseFee())
+		tx, err := types.SignTx(types.NewTx(&types.BlobTx{
+			Nonce:      0,
+			GasTipCap:  uint256.NewInt(1),
+			GasFeeCap:  uint256.MustFromBig(fee),
+			Gas:        params.TxGas,
+			To:         testEthAddrs[0],
+			BlobFeeCap: uint256.NewInt(1),
+			BlobHashes: []common.Hash{{1}},
```

Comment:
(nit) I think it's worth putting a comment pointing out that this is what's important. Like:

```go
BlobHashes: []common.Hash{{1}}, // <----- The existence of this blob MUST cause an error
```

Reply:
will apply this in a new PR to avoid voiding the approvals here.